http-common/src/connector.rs

```diff
@@ -172,45 +66,92 @@ impl Connector {
 
 	#[cfg(feature = "tokio02")]
 	pub async fn incoming(self) -> std::io::Result<Incoming> {
-		match self {
-			Connector::Fd { fd, .. } => {
-				let sock_addr = nix::sys::socket::getsockname(fd).map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err))?;
-				match sock_addr {
-					nix::sys::socket::SockAddr::Inet(_) => {
-						let listener = unsafe { std::os::unix::io::FromRawFd::from_raw_fd(fd) };
-						let listener = tokio::net::TcpListener::from_std(listener)?;
+		let systemd_socket = get_systemd_socket().map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err))?;
+		if let Some(fd) = systemd_socket {
+			let sock_addr = nix::sys::socket::getsockname(fd).map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err))?;
+			match sock_addr {
+				nix::sys::socket::SockAddr::Inet(_) => {
```

Comment:
Do we support TCP socket configuration for systemd for release builds?

Reply:
Good point. Fixed.